### ISSUE TITLE
Unable to Create Any Builds with `eas build`

### ISSUE THREAD

**Issue by NLanese** [expo/expo-cli#4379]
### Summary

While my app runs perfectly on my ios Simulator, iPhone, and Android Device, whenever I run `eas build` it fails for both options. 
I almost always fail every android build the first time because there appears to be no limit to the things that can screw up the `run gradlew` process, but this time I am currently out of fixes.
Furthermore, now my ios build will not complete, mentioning Podspecs and Podfile.lock which are both files I've never created or worked in, so I can only assume this is something done on expo's side.

### Environment

System:
      OS: macOS 12.3.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 18.0.0 - /usr/local/bin/node
      Yarn: 1.22.18 - /usr/local/bin/yarn
      npm: 8.6.0 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^45.0.0 => 45.0.4 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
      react-navigation: ^4.4.4 => 4.4.4 
    npmGlobalPackages:
      eas-cli: 0.50.0
      expo-cli: 5.0.3
    Expo Workflow: bare

### Please specify your device/emulator/simulator platform, model and version

N/A

### Error output

ios: 
Error: Compatible versions of some pods could not be resolved.
You are seeing this error because either:
  - Versions in the Podfile.lock cached by EAS do not match required values in Podspecs of some of the libraries. To fix that add the "cache.key" field (it can be set to any value) in eas.json to invalidate the cache.
  - Some of the pods used in your project depend on different versions of the same pod. Please see logs for more info.

android:
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0
Build failed: Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.

### Reproducible demo or steps to reproduce from a blank project

Run eas build and follow the instructions laid out, using the most recent sdk

**Comment by danieldanielecki** [comment]
What type of errors are you getting?
I'm getting an obvious one regarding `google-services.json`, and indeed that's only on `eas`.
Also on the same step

```
[stderr] Execution failed for task ':app:processReleaseGoogleServices'.
[stderr] > File google-services.json is missing. The Google Services Plugin cannot function without it.
```
I've checked 100x if the file is there... no problem with `expo build:android`.

My build ID is `a72dae79-5f10-4a28-bf75-2e5d6d5ea502`.

**Comment by NLanese** [comment]
I'm getting way too many on the Android Build to even respond to here, pretty much everything between lines 30-40 is an error, and every task fails after line 434 (all the way to 848, so I clearly can't copy and paste that much haha). 
As far as the apple build, it was just 
```
Build failed: Compatible versions of some pods could not be resolved.
You are seeing this error because either:
  - Versions in the Podfile.lock cached by EAS do not match required values in Podspecs of some of the libraries. To fix that add the "cache.key" field (it can be set to any value) in eas.json to invalidate the cache.
  - Some of the pods used in your project depend on different versions of the same pod. Please see logs for more info.
```
Which I hadn't seen  my first few builds, so I'm not exactly sure what's going on so much differently right now

**Comment by brentvatne** [comment]
> What type of errors are you getting? I'm getting an obvious one regarding `google-services.json`, and indeed that's only on `eas`. Also on the same step
> 
> ```
> [stderr] Execution failed for task ':app:processReleaseGoogleServices'.
> [stderr] > File google-services.json is missing. The Google Services Plugin cannot function without it.
> ```
> 
> I've checked 100x if the file is there... no problem with `expo build:android`.
> 
> My build ID is `a72dae79-5f10-4a28-bf75-2e5d6d5ea502`.

only files included in your git archive are uploaded, see more info at https://expo.fyi/eas-build-archive - you can treat EAS Build as other CI services - your project is downloaded to the machine and run there.

**Comment by brentvatne** [comment]
> Versions in the Podfile.lock cached by EAS do not match required values in Podspecs of some of the libraries. To fix that add the "cache.key" field (it can be set to any value) in eas.json to invalidate the cache.

@NLanese - did you try following these instructions? can you link me to the related build jobs?

for your android builds, did you read the build logs for more information? can you share a link to the build pages? here's a guide you can follow for troubleshooting: https://docs.expo.dev/build-reference/troubleshooting/

**Comment by danieldanielecki** [comment]
> > What type of errors are you getting? I'm getting an obvious one regarding `google-services.json`, and indeed that's only on `eas`. Also on the same step
> > ```
> > [stderr] Execution failed for task ':app:processReleaseGoogleServices'.
> > [stderr] > File google-services.json is missing. The Google Services Plugin cannot function without it.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I've checked 100x if the file is there... no problem with `expo build:android`.
> > My build ID is `a72dae79-5f10-4a28-bf75-2e5d6d5ea502`.
> 
> only files included in your git archive are uploaded, see more info at https://expo.fyi/eas-build-archive - you can treat EAS Build as other CI services - your project is downloaded to the machine and run there.

Wow, what a finding!

It was the case, indeed. The only way not to commit `google-service.json` would be to encode with `base64`??

**Comment by NLanese** [comment]
@brentvatne Yes, although it is likely I did not do this step perfectly, I got the following error after I added `Error: eas.json is not valid [ValidationError: "cache.key" is not allowed]`

I tried to follow along with the build logs, but there wasn't much helpful information aside from the `cache.key` suggestion which unfortunately failed. The link to my builds is here: https://expo.dev/accounts/nlanese/projects/the-tom-app/builds/1ac5fd64-b0fb-4112-b647-fc3274aeb229

**Comment by brentvatne** [comment]
@NLanese - here's more info on cache key: https://docs.expo.dev/build-reference/eas-json/#cache
